File: src/backend/models/DonationModel.py
from dataclasses import dataclass
from datetime import datetime

@dataclass
class DonationModel:
    Id: int
    Username: slice
    Amount: slice
    Notes: slice
    Method: slice
    DateAdded: datetime
    
def data_to_model(data):
    if not data:
        return None
    return DonationModel(
        Id=data.get('Id'),
        Username=data.get('Username'),
        Amount=data.get('Amount'),
        Notes=data.get('Notes'),
        Method=data.get('Method'),
        DateAdded=data.get('DateAdded')
    )

File: src/app/models/donationmodel.ts
export class DonationModel {
    Id = <number>(0);
    Username = <string>("");
    Amount = <string>("");
    Notes = <string>("");
    Method = <string>("");
    DateAdded = <string>("");

    constructor(init?: Partial<DonationModel>) {
        Object.assign(this, init);
    }

    assignData(data: DonationModel) {
        this.Id = data.Id;
        this.Username = data.Username;
        this.Amount = data.Amount;
        this.Notes = data.Notes;
        this.Method = data.Method;
        this.DateAdded = data.DateAdded;
    }

    getData() {
        return {
            Id: this.Id,
            Username: this.Username,
            Amount: this.Amount,
            Notes: this.Notes,
            Method: this.Method,
            DateAdded: this.DateAdded
        };
    }
}
